### PR TITLE
Remove Record suffix

### PR DESCRIPTION
**Pull request by sifnoc**
- #358 item (4) - Remove Record suffix
    - ValidatorRecord => Validator
    - CrosslinkRecord => Crosslink
    - ~ShardReassignmentRecord => ShardReassignment~ removed from #382 
    - PendingAttestationRecord => PendingAttestation

My first contribution, so Please do not hesitate to point my errors.

### CODE REVIEW

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -43,11 +43,11 @@
...
         - [Beacon chain state](#beacon-chain-state)
             - [`BeaconState`](#beaconstate)
-            - [`ValidatorRecord`](#validatorrecord)
-            - [`CrosslinkRecord`](#crosslinkrecord)
+            - [`Validator`](#Validator)
+            - [`Crosslink`](#Crosslink)
             - [`ShardCommittee`](#shardcommittee)
             - [`DepositRootVote`](#depositrootvote)
-            - [`PendingAttestationRecord`](#pendingattestationrecord)
+            - [`PendingAttestation`](#PendingAttestation)
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: Should be `#pendingattestation` (no capitals)

Comment (sifnoc) [reply specs/core/0_beacon-chain.md]: Thanks for a correction.

Comment (sifnoc) [reply specs/core/0_beacon-chain.md]: Could you explain to write no capital? it works following header with A capital.
Anyway, Thank you for review and Point my errors.

Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: For consistency, I guess. Putting capitals everywhere works also.

File: specs/core/0_beacon-chain.md [hunk 2]
```diff
@@ -43,11 +43,11 @@
...
         - [Beacon chain state](#beacon-chain-state)
             - [`BeaconState`](#beaconstate)
-            - [`ValidatorRecord`](#validatorrecord)
-            - [`CrosslinkRecord`](#crosslinkrecord)
+            - [`Validator`](#Validator)
+            - [`Crosslink`](#Crosslink)
             - [`ShardCommittee`](#shardcommittee)
             - [`DepositRootVote`](#depositrootvote)
-            - [`PendingAttestationRecord`](#pendingattestationrecord)
+            - [`PendingAttestation`](#Pendingattestation)
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: `#Pendingattestation` => `#pendingattestation`

File: specs/core/0_beacon-chain.md (hunk 1 above)
Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: The generator that @hwwhww uses defaults to lower case. ++ for consistency.

Comment (sifnoc) [reply specs/core/0_beacon-chain.md]: Changed all lower case, I will try to check consistency next PR.

File: specs/core/0_beacon-chain.md [hunk 3]
```diff
@@ -878,8 +878,9 @@ def get_committee_count_per_slot(active_validator_count: int) -> int:
 
 ```python
 def get_shuffling(randao_mix: Hash32,
-                  validators: List[ValidatorRecord],
-                  slot: int) -> List[List[int]]
+                  validators: List[Validator],
+                  crosslinking_start_shard: int,
```
[the remark this answers is not in the document]
Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: no, `get_shuffling` just returns the shuffling split across slots.
`get_shard_committees_at_slot` returns an array of arrays of (validators, shard) tuples.

Arguably, the naming and return value of these two functions is not great at this point. Maybe `get_shuffling` should just return an array of shuffled validator indices and the `get_shard_committees_at_slot` should do all of the splitting.

Don't feel super strongly about this.

File: specs/core/0_beacon-chain.md (hunk 1 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: Oh, which generator? I want :)

Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: It's a vscode plugin: https://github.com/AlanWalk/markdown-toc
but I don't use vscode..

File: specs/core/0_beacon-chain.md (hunk 3 above)
Comment (djrtwo) [inline specs/core/0_beacon-chain.md]: I think this `crosslinking_start_shard` got added in a bad merge or something. Remove this line

**Comment by sifnoc** [comment]
I could understand what I can do for contributing through this PR.
Thanks for justin, djrtwo.

**Pull request merged**